### PR TITLE
Migrate obs-infraobs-integrations to package-spec v3 #4

### PR DESCRIPTION
**Pull request by shmsr**
Thanks to this PR https://github.com/elastic/integrations/pull/8050 and @tommyers-elastic , could migrate 4 more packages

- [x] azure_app_service
- [x] citrix_adc
- [x] redis
- [x] tomcat

ECS dictates that that certain fields (geo, interface, os) should be nested and package validations were failing with these fields defined at root level. So from ecs.yml they are moved to fields.yml

Also refer: https://www.elastic.co/guide/en/ecs/current/ecs-field-reference.html

## Proposed commit message

Migrate the following package to use package-spec v3:

- [x] azure_app_service
- [x] citrix_adc
- [x] redis
- [x] tomcat

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/redis/data_stream/info/fields/fields.yml [hunk 1]
```diff
@@ -524,11 +524,13 @@
...
       description: |
         Count of slow operations
-- name: service.address
-  type: keyword
-  # Reason for adding as dimension field : A connection string to the server.
-  dimension: true
-  description: Client address
-- name: service.version
-  type: keyword
-  description: Version of the service the data was collected from
```
Comment (shmsr) [inline packages/redis/data_stream/info/fields/fields.yml:534]: Defined in ecs.yml

File: packages/redis/data_stream/info/fields/ecs.yml [hunk 2]
```diff
@@ -10,6 +8,7 @@
   name: ecs.version
 - external: ecs
   name: service.address
+  # Reason for adding as dimension field : A connection string to the server.
```
Comment (shmsr) [inline packages/redis/data_stream/info/fields/ecs.yml:11]: Copied from fields.yml

File: packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml [hunk 3]
```diff
@@ -129,3 +129,34 @@
     - name: log
       type: keyword
       description: Details about the event depending on level
+# The following `geo` mappings are not allowed at root level in ECS, and are defined here for backward compatibility.
```
Comment (tommyers-elastic) [inline packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml:132]: please add an item to the list of "Other non-blocking issues that should be fixed in the future" in the tracking issue to remove these later.

File: packages/citrix_adc/data_stream/interface/fields/fields.yml [hunk 4]
```diff
@@ -227,3 +227,11 @@
                   description: Number of bytes transmitted by an interface since the NetScaler appliance was started or the interface statistics were cleared.
                   metric_type: counter
                   unit: byte
+# The following `interface` mappings are not allowed at root level in ECS, and are defined here for backward compatibility.
```
Comment (tommyers-elastic) [inline packages/citrix_adc/data_stream/interface/fields/fields.yml:230]: same here

File: packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml (hunk 3 above)
Comment (tommyers-elastic) [reply packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml:132]: are these fields also copied to a valid location e.g. `client.geo` or `source.geo`? if not please also add an item to that list to do that too.

**Review by tommyers-elastic — APPROVED** [review]
awesome - LGTM. thanks!

File: packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml (hunk 3 above)
Comment (shmsr) [reply packages/azure_app_service/data_stream/app_service_logs/fields/fields.yml:132]: done.

**Pull request merged**